core/policy.go

```diff
@@ -50,28 +50,28 @@ func NewMessageQueuePolicy(store chain.BlockProvider, maxAge uint64) *DefaultQue
 
 // HandleNewHead updates the policy target in response to a new head tipset.
 func (p *DefaultQueuePolicy) HandleNewHead(ctx context.Context, target PolicyTarget, oldHead, newHead types.TipSet) error {
-	_, newBlocks, err := CollectBlocksToCommonAncestor(ctx, p.store, oldHead, newHead)
+	_, newTips, err := CollectTipsToCommonAncestor(ctx, p.store, oldHead, newHead)
 	if err != nil {
 		return err
 	}
 
 	// Remove from the queue all messages that have now been mined in new blocks.
 
-	// Rearrange the blocks in increasing height order so messages are discovered in order.
-	// Note: this is imperfect until CollectBlocksToCommonAncestor is updated to return blocks
```

Comment:
What about this note?  Is it still relevant, has it been fixed, or is there a story?

Reply:
I didn't realize the "Note" part was part of it, ok.